framework/SupervisedLearning/ARMA.py

```diff
@@ -254,14 +276,29 @@ def __trainLocal__(self,featureVals,targetVals):
         self._trainingCDF[target] = mathUtils.trainEmpiricalFunction(timeSeriesData, minBins=self._minBins)
       # if this target governs the zero filter, extract it now
       if target == self.zeroFilterTarget:
-        self.notZeroFilterMask = self._trainZeroRemoval(timeSeriesData,tol=self.zeroFilterTol) # where zeros are not
-        self.zeroFilterMask = np.logical_not(self.notZeroFilterMask) # where zeroes are
+        self.notZeroFilterMask = self._trainZeroRemoval(timeSeriesData,tol=self.zeroFilterTol) # where zeros or less than zeros are
+        self.zeroFilterMask = np.logical_not(self.notZeroFilterMask) # where data are
       # if we're removing Fourier signal, do that now.
+      maskRes = timeSeriesData == timeSeriesData
+
+      if target in self.peaks:
```

Comment:
Would you mind adding a few comments here explaining the thoughts behind the process?